### DISCUSSION TITLE
Edges example with custom edges not working

### DISCUSSION THREAD

**Discussion by mahamzafar979** [bcakmakoglu/vue-flow#373] [Q&A] [answered]
Hi,

I tried replicating the edges example from below link into my project.
https://vueflow.dev/examples/edges/
After replicating, CustomEdges were not working properly as shown in below image.
![image](https://user-images.githubusercontent.com/73057067/197652017-f0ca76ec-d468-4d7f-a89c-b126b029d985.png)

and on console I am getting this error: 
Error: <path> attribute d: Unexpected end of attribute. Expected number, "M".
On debugging I noticed that getBezierPath function is returning a string instead of an array that I think is causing the issue.
I am looking into custom edges as I want to show multiline text as edges label. 
Can someone help me with that?

Thanks.

**Comment by bcakmakoglu** [comment, marked as answer]
What version are you using? The change of `getBezierPath` occured in 1.0.0 (see [here](https://github.com/bcakmakoglu/vue-flow/blob/master/packages/core/CHANGELOG.md#100))

The examples always use the version that is shown in the header (currently 1.2.0 is the latest version and the one used on the page). So if you want to replicate the code, you would do best to use that version :)

**Reply by bcakmakoglu** [reply]
Also a small note:
Edge labels are kinda limited in what they can do - in a lot of cases you'll have an easier time using additional nodes that intercept edges and pose as labels.

**Reply by mahamzafar979** [reply]
I am using 1.2.0 version for vue-flow/core and 1.1.0 for additional components

**Reply by bcakmakoglu** [reply]
Can you provide a reproduction?
Seems to work fine for me. Here's a working sandbox: https://codesandbox.io/s/serverless-hooks-ji3e21?file=/src/components/CustomEdge2.vue:924-925

**Reply by mahamzafar979** [reply]
Hi,
While trying to reproduce the issue I found out that the issue was with the nuxt3 version I was using. On changing the version example is working now.

**Reply by mahamzafar979** [reply]
I have one more question:
By default, labels are placed at the center of an edge but as I am assigning dynamic x & y coordinates to nodes. Hence, if the labels are placed in the middle then it's difficult to determine which link the label belongs to or if they are placed close together. Please refer to the screenshot below
![image](https://user-images.githubusercontent.com/73057067/197862759-3e7e0248-d6c1-4663-bc48-21038e0b452c.png)
So is there a way to place the labels closer to the target node on the link?

**Reply by bcakmakoglu** [reply]
Sorry I didn't see your last message.

So clarify your question: Of course you can put it closer to the nodes.
The utilities like `getBezierPath` or `getSmoothStepPath` will return an array like `[path, labelX, labelY]` you can use these values to create a custom edge and a label. X and Y are the center values, so if you need it closer into some direction just add or subtract some amount to move the label into your desired direction.

Check the [edges example](https://vueflow.dev/examples/edges/) again to see how it works and play around with it a bit :)

Note: There have been some recent updates, like using a `div` based edge label, which you can find in the example as well.

If you have more questions, let me know :)
